sormas-backend/src/main/java/de/symeda/sormas/backend/region/CountryFacadeEjb.java

```diff
@@ -0,0 +1,223 @@
+
+@Stateless(name = "CountryFacade")
+public class CountryFacadeEjb implements CountryFacade {
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	@EJB
+	private CountryService countryService;
+
+	@Override
+	public CountryDto getCountryByUuid(String uuid) {
+		return toDto(countryService.getByUuid(uuid));
+	}
+
+	@Override
+	public List<CountryReferenceDto> getByDefaultName(String name, boolean includeArchivedEntities) {
+		return countryService.getByDefaultName(name, includeArchivedEntities).stream().map(r -> toReferenceDto(r)).collect(Collectors.toList());
+	}
+
+	@Override
+	public List<CountryIndexDto> getIndexList(CountryCriteria criteria, Integer first, Integer max, List<SortProperty> sortProperties) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Country> cq = cb.createQuery(Country.class);
+		Root<Country> country = cq.from(Country.class);
+
+		Predicate filter = countryService.buildCriteriaFilter(criteria, cb, country);
+
+		if (filter != null) {
+			cq.where(filter).distinct(true);
+		}
+
+		if (sortProperties != null && sortProperties.size() > 0) {
+			List<Order> order = new ArrayList<Order>(sortProperties.size());
+			for (SortProperty sortProperty : sortProperties) {
+				Expression<?> expression;
+				switch (sortProperty.propertyName) {
+				case Country.DEFAULT_NAME:
+				case Country.EXTERNAL_ID:
+				case Country.ISO_CODE:
+				case Country.UNO_CODE:
+					expression = country.get(sortProperty.propertyName);
+					break;
+				default:
+					throw new IllegalArgumentException(sortProperty.propertyName);
+				}
+				order.add(sortProperty.ascending ? cb.asc(expression) : cb.desc(expression));
+			}
+			cq.orderBy(order);
+		} else {
+			cq.orderBy(cb.asc(country.get(Country.ISO_CODE)));
+		}
+
+		cq.select(country);
+
+		if (first != null && max != null) {
+			return em.createQuery(cq)
+				.setFirstResult(first)
+				.setMaxResults(max)
+				.getResultList()
+				.stream()
+				.map(f -> toIndexDto(f))
+				.collect(Collectors.toList());
+		} else {
+			return em.createQuery(cq).getResultList().stream().map(f -> toIndexDto(f)).collect(Collectors.toList());
+		}
+	}
+
+	@Override
+	public long count(CountryCriteria criteria) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Long> cq = cb.createQuery(Long.class);
+		Root<Country> root = cq.from(Country.class);
+
+		Predicate filter = countryService.buildCriteriaFilter(criteria, cb, root);
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+
+		cq.select(cb.count(root));
+		return em.createQuery(cq).getSingleResult();
+	}
+
+	@Override
+	public String saveCountry(CountryDto dto) throws ValidationRuntimeException {
+		if (StringUtils.isBlank(dto.getIsoCode())) {
+			throw new EmptyValueException(I18nProperties.getValidationError(Validations.importCountryEmptyIso));
+		}
+
+		Country country = countryService.getByUuid(dto.getUuid());
+
+		if (country == null
```

Comment:
@vlad-ciucescu It seems unnecessary and unperformant to do another query to the DB here. There are DB constraints in place that should do the check.

Reply:
Normally I would agree but

- we will at most have ~200 countries to import, unlike cases for example where there can be thousands. When there are so few I think the impact will be insignificant.
- this is also done in other `save` methods, for example for regions or even in cases, in the `validate` method
- this is also useful for specifically throwing the `ValidationRuntimeException` which can then be caught in the Importer. Otherwise, it gets much clumsier with the `TransactionRolledbackException` or whatever is thrown.

I would prefer to leave it like this.